### ISSUE TITLE
[v5] If you have https enabled for your URL invite user links is still generated with http

### ISSUE THREAD

**Issue by chhatbarjignesh** [reportportal/reportportal#1019]
**Describe the bug**
If you have https enabled for your URL and try to invite the user the link that it creates is with HTTP:// not https:// and due to that when a user tries to create an account it says site can't be reached

**To Reproduce**
deploy rp with https enabled
enable email plugin
invite user using email

**Expected behavior**
It should allow the user to create the user

**Screenshots**
![image](https://user-images.githubusercontent.com/34551874/84925313-b61b1580-b097-11ea-8493-195cd471f4e9.png)

RP version 5.2.2

**Comment by evjlobanova** [comment]
@chhatbarjignesh Could you please send a screenshot with invite form 
![image](https://user-images.githubusercontent.com/30413511/84998041-fb068100-b157-11ea-9f76-80139f3197cc.png)

**Comment by chhatbarjignesh** [comment]
@evjlobanova INvite link is also http only
![image](https://user-images.githubusercontent.com/34551874/85159100-0aa2ca00-b22b-11ea-838c-d57c7e5e2ab7.png)

**Comment by evjlobanova** [comment]
@Yumfriez where can be a root cause for such behavior?

**Comment by e-olga** [comment]
@evjlobanova 
The issue is not reproduced on our environments with 5.2.2 version.
In case the RP works on https://, the invitation link created with https:// as well

@Yumfriez ,
are the any ideas?

**Comment by chhatbarjignesh** [comment]
How are you determining the logic that your site is http:// or https:// I have installed certificate at LB level in AWS. Just in case it helps you to quickly figure out the root cause of the issue

**Comment by Yumfriez** [comment]
@evjlobanova Probably we need help from @Cyberglamdring to know about our certificates configs

**Comment by IharBletskoTR** [comment]
Same issue - RP behind https loadbalancer and links from RP are sent with "http", instead of "https".
Option to select "auto/http/https" for links would be good enough to solve this.

**Comment by elavrenchuk** [comment]
The same thing here. 
RP is behind nginx proxy with SSL configured. Invitations and password restoring emails are got to the end users with broken links:

![image](https://user-images.githubusercontent.com/30771249/100113456-ff5a6700-2e78-11eb-8c47-b3e81f8519de.png)

![image](https://user-images.githubusercontent.com/30771249/100113667-3df02180-2e79-11eb-9e94-4e07503f8813.png)

**Comment by IharBletskoTR** [comment]
@elavrenchuk 
Look here, fix is on the way (but at next year) https://github.com/reportportal/reportportal/issues/1210
As a workaround you may try to do http-to-https redirecting in nginx.

**Comment by elavrenchuk** [comment]
Might be useful for those who are using nginx as a proxy.

Just add this to your site config, in ssl section. 
`ssl off; `

In general, config might have next context:

```
server {
    listen 80;
    server_name reportportal.example.com;
    return 301 https://$host$request_uri;
}

server {
    listen 443 ssl;
    server_name reportportal.example.com;

    ssl off;
    ssl_certificate /etc/ssl/web/your-cert.crt;
    ssl_certificate_key /etc/ssl/web/your-cert.rsa;

    location / {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_pass http://127.0.0.1:8080;
        proxy_http_version 1.1;
        proxy_request_buffering off;
        proxy_buffering off;
    }
}
```
Then links in invites will be in the right manner:
![image](https://user-images.githubusercontent.com/30771249/100194008-979a2f80-2efd-11eb-92c2-694f436e52e7.png)

**Comment by evjlobanova** [comment]
We will add the possibility to choose a host explicitly.

**Comment by evjlobanova** [comment]
Started from version 5.4 a feature will be available 
![image](https://user-images.githubusercontent.com/30413511/101609794-42a9ef00-39bc-11eb-934c-50d1a705dfd6.png)

**Comment by chhatbarjignesh** [comment]
@evjlobanova, I have added rpHost as integration parameters as suggested, and I am on v5.5.0, but when I try to reset the password using forgot password, the email I receive has the link with http:// and not https://

Would you mind letting me know if I have missed anything in configuration?

API Response of the email server configuration:
 ```
{
    "id": 29,
    "name": "email server",
    "integrationType": {
      "type": 2,
      "name": "email",
      "enabled": true,
      "creationDate": 1578931566104,
      "groupType": "NOTIFICATION"
    },
    "integrationParameters": {
      "protocol": "smtp",
      "rpHost": "https://reportportal.******.*****.com",
      "authEnabled": true,
      "port": "465",
      "sslEnabled": false,
      "starTlsEnabled": false,
      "host": "appmail.private.******.com",
      "username": "*****@******.com"
    },
    "enabled": true,
    "creator": "jchhatbar",
    "creationDate": 1591392673703
  },
```

email: screenshot

![image](https://user-images.githubusercontent.com/34551874/128899235-34380e39-2946-46c4-a726-c7c6000d48b8.png)

**Comment by evjlobanova** [comment]
@Yumfriez could you please check this issue?

**Comment by evjlobanova** [comment]
@VolhaKarenko could you please add a new bug to the next sprint on Back End

**Comment by evjlobanova** [comment]
Will be fixed in version 5.6
